src/react/utils.js

```diff
@@ -264,3 +263,35 @@ export function normalizeFunctionalComponentParamaters(func: ECMAScriptSourceFun
     }
   });
 }
+
+export function getThisAssignments(functionBody: BabelNodeBlockStatement): Set<string> {
+  let assignments = new Set();
+  traverse(
+    t.file(t.program([t.functionDeclaration(t.identifier("dummy"), [], functionBody)])),
+    {
+      "ThisExpression|Identifier": function(path) {
+        if (t.isThisExpression(path.node) || path.node.name === "this") {
+          let parentPath = path.parentPath;
+          let parentNode = parentPath.node;
+
+          if (parentPath && t.isMemberExpression(parentNode) && parentPath.parentPath) {
+            let topNode = parentPath.parentPath.node;
+            if (t.isAssignmentExpression(topNode) && topNode.operator === "=") {
```

Comment:
There are other operators like `+=` that are also assignment. Do they not count?

Reply:
@sebmarkbage You'd need to create the assignment to a value before incrementing right?

Reply:
Technically you don't need to, but I guess it's most likely that you would.